localstack-core/localstack/services/s3/utils.py

```diff
@@ -516,8 +517,8 @@ def uses_host_addressing(headers: Dict[str, str]) -> str | None:
         return bucket_name
 
 
-def get_class_attrs_from_spec_class(spec_class: Type[str]) -> set[str]:
-    return {getattr(spec_class, attr) for attr in vars(spec_class) if not attr.startswith("__")}
+def get_class_attrs_from_spec_class(spec_class: type[StrEnum]) -> set[str]:
+    return {str(spec) for spec in spec_class}
```

Comment:
This is the only actual code change necessary, which iterates over the classes, now an enum.